commit bc6bc9fac98074ce963353ef5f1ab445d1e01b91
Author: Evan Stade <estade@chromium.org>
Date:   Mon Aug 22 20:54:52 2022 +0000

    Fix some style issues in AutofillPopupViewPtr
    
    Bespoke memory management like this should probably be avoided, but
    without delving too deeply, we can at least:
    
    * remove two single-arg constructors that are being used for implicit
      conversion
    * disallow copy and assign
    * make use of default ctor
    
    This has the handy side-effect of removing a reference to
    std::nullptr_t that wasn't qualified with std::.
    
    Bug: 957519
    Change-Id: I3934f037892da8a14eb101b864386594d1d8eb7c
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3846070
    Commit-Queue: Evan Stade <estade@chromium.org>
    Reviewed-by: Christoph Schwering <schwering@google.com>
    Cr-Commit-Position: refs/heads/main@{#1037932}

--- a/chrome/browser/ui/autofill/autofill_popup_controller_impl.h
+++ b/chrome/browser/ui/autofill/autofill_popup_controller_impl.h
@@ -178,8 +178,13 @@ class AutofillPopupControllerImpl : publ
   class AutofillPopupViewPtr {
    public:
     AutofillPopupViewPtr() = default;
-    AutofillPopupViewPtr(nullptr_t) : ptr_(nullptr) {}
-    AutofillPopupViewPtr(AutofillPopupView* ptr) : ptr_(ptr) {}
+    AutofillPopupViewPtr(const AutofillPopupViewPtr&) = delete;
+    AutofillPopupViewPtr& operator=(const AutofillPopupViewPtr&) = delete;
+
+    AutofillPopupViewPtr& operator=(AutofillPopupView* ptr) {
+      ptr_ = ptr;
+      return *this;
+    }
 
     explicit operator bool() const { return ptr_; }
 
@@ -228,7 +233,7 @@ class AutofillPopupControllerImpl : publ
 
   PopupControllerCommon controller_common_;
   raw_ptr<content::WebContents> web_contents_;
-  AutofillPopupViewPtr view_ = nullptr;  // Weak reference.
+  AutofillPopupViewPtr view_;
   base::WeakPtr<AutofillPopupDelegate> delegate_;
 
   // If set to true, the popup will never be hidden because of stale data or if
